### PR TITLE
Cast federation payload explicilty to an object

### LINKED ISSUE #685: Federated multi-search example does not work
**Description**
I'm using the federated multi-search feature of Meilisearch. I use the code example from the docs:

```php
$client->multiSearch([
    (new SearchQuery())
      ->setIndexUid('movies'))
      ->setQuery('batman'),
    (new SearchQuery())
      ->setIndexUid('comics')
      ->setQuery('batman'),
  ],
  (new MultiSearchFederation())
);
```
However, this results in the following error:

afbeelding

This error **only** appears when I pass an empty MultiSearchFederation() instance, like in the docs. The error **does not** appear when I set a property of MultiSearchFederation, like `(new MultiSearchFederation())->setLimit(limit: 100)`.

**Possible solution**
See my PR

### PR DESCRIPTION
**Pull request by thijskuilman**
# Pull Request

## Related issue
Fixes #685

## What does this PR do?
- It casts the `federation` payload explicitly to an object. This solves the error that appears when no `MultiSearchFederation` properties have been set. 

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

### CODE REVIEW

**Comment by norkunas** [comment]
Please add test then to avoid regressions in future

**Review by curquiza — CHANGES_REQUESTED** [review]
@thijskuilman 

thank you for your PR ❤️

I agree with norkunas, can you add a test with an empty MultiSearchFederation to ensure it works all the time?

**Comment by thijskuilman** [comment]
@norkunas @curquiza I have added a test!

**Comment by norkunas** [comment]
@thijskuilman not sure that we need a functional test here :) better would be to assert on that what was really passed to http client, because now a test without assertions doesn't prove anything

**Comment by norkunas** [comment]
IMHO we should a dev requirement for symfony/http-client, which allows easier testing than the guzzle.

Then your test could look like this:

```
$mockHttpClient = new \Symfony\Component\HttpClient\MockHttpClient(static function (string $method, string $url, array $options) {
    // assert on method, url, options['body'] or whatever what was passed

    return new MockResponse();
});
$httpClient = new \Symfony\Component\HttpClient\Psr18Client($mockHttpClient);

$meilisearch = new \Meilisearch\Client('http://meilisearch', 'api_key', $httpClient);
$meilisearch->multiSearch([], new MultiSearchFederation());
```

code is just an idea and is not validated :)

**Comment by norkunas** [comment]
My point was that unit test will be faster than the functional one :)

File: src/Endpoints/Delegates/HandlesMultiSearch.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function multiSearch(array $queries = [], ?MultiSearchFederation $federat
 
         $payload = ['queries' => $body];
         if (null !== $federation) {
-            $payload['federation'] = $federation->toArray();
+            $payload['federation'] = (object) $federation->toArray();
```
Comment (ruhulfbr) [inline src/Endpoints/Delegates/HandlesMultiSearch.php:28]: Why is typecasting necessary? You could use `$payload['federation'] = $federation;` and it would work just fine.

**Comment by ruhulfbr** [comment]
@curquiza  Could you please review and process this PR or provide a solution? This is an important update.

**Comment by curquiza** [comment]
Hello @ruhulfbr 
we have a lot on our plate now, but @Strift will review your PR when he will have more time for this
In the meantime if you have time @norkunas, you can review and merge of course. If you cannot, no worries, we will take care of it later

**Comment by norkunas** [comment]
I've already reviewed and my point still stands - it'd enough for unit test

**Comment by norkunas** [comment]
Closing as it's already been fixed
